From f2f105f518413ea3e4c212f89585f9a8a5dddcdd Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Thu, 4 Nov 2021 14:11:02 +1030
Subject: [PATCH 57/97] PR28540, segmentation fault on NULL byte_get

	PR 28540
	* objdump.c (dump_bfd): Don't attempt load_separate_debug_files
	when byte_get is NULL.
---
 binutils/objdump.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/binutils/objdump.c b/binutils/objdump.c
index 356e3ecd853..1bfb5c71e7a 100644
--- a/binutils/objdump.c
+++ b/binutils/objdump.c
@@ -4862,7 +4862,7 @@ dump_bfd (bfd *abfd, bool is_mainfile)
      
      The test on is_mainfile is there because the chain of separate debug
      info files is a global variable shared by all invocations of dump_bfd.  */
-  if (is_mainfile)
+  if (byte_get != NULL && is_mainfile)
     {
       load_separate_debug_files (abfd, bfd_get_filename (abfd));
 
-- 
2.34.0

